### ISSUE TITLE
Add support for custom cluster definition (as part of device definition)

### ISSUE THREAD

**Issue by LaurentChardin** [nurikk/z2m-frontend#2001]
**Is your feature request related to a problem? Please describe.**
There is an ongoing development in z2m to enable custom cluster definition at the device definition and have them decoupled from the zh lib.

However as [sjorge](https://github.com/sjorge) pointed out, the z2m-frontend also has a static dependency on those libs. Once the change lands in z2m and people start to migrate cluster definitions out of **zh**, they will loost the visibility in the UI:
- **_In the dev tab_**, the cluster dropdown would not contain the custom cluster name nor the cluster ID 

_This request is to start a change in order to support the upcoming changes_ : https://github.com/Koenkk/zigbee-herdsman/pull/1019

**Describe the solution you'd like**
In the future, there will be standard cluster definition in ZH (+legacy ones) and new cluster definitions in ZHC (carried directly at the device level).

We can either:
- Implement the same logic than z2m in merging the definitions depending on the device currently used
- Remove completely the dependency on zh/zhc and request the device definition on the fly to z2m (and let him do the merge) -> preferred option

**Comment by Suxsem** [comment]
Hi, any progress on this one? Thank you!

**Comment by LaurentChardin** [comment]
@nurikk Hope if you could help with the latest commit : at least to trigger the build.

**Comment by LaurentChardin** [comment]
Merged into `0.7.0`
